src/components/Map/EditableMapDraw.tsx

```diff
@@ -0,0 +1,229 @@
+
+      setDrawMode(initialMode);
+      setMapRef(map);
+
+      return new MapboxDraw({ defaultMode: initialMode, ...defaultProps, ...otherProps });
+    },
+    ({ map }) => {
+      // @ts-ignore
+      map.off('draw.create', onCreate);
+      // @ts-ignore
+      map.off('draw.delete', onDelete);
+      // @ts-ignore
+      map.off('draw.modechange', onModeChange);
+      // @ts-ignore
+      map.off('draw.update', onUpdate);
+    }
+  );
+
+  useEffect(() => {
+    if (setMode && initializedGeometry) {
+      const features = draw.getAll().features;
+      const hasFeature = features && featureHasCoordinates(features[0]);
+
+      if (drawMode === 'draw_polygon') {
+        if (hasFeature) {
+          setMode('ReplacingBoundary');
+        } else {
+          setMode('CreatingBoundary');
+        }
+      } else if (drawMode === 'simple_select') {
+        if (selection) {
+          setMode('BoundarySelected');
+        } else if (hasFeature) {
+          setMode('BoundaryNotSelected');
+        } else {
+          setMode('NoBoundary');
+        }
+      } else if (drawMode === 'direct_select') {
+        if (selection) {
+          setMode('EditingBoundary');
+        } else {
+          setMode('BoundaryNotSelected');
+        }
+      }
+    }
+  }, [draw, drawMode, initializedGeometry, selection, setMode]);
+
+  const removeExistingPolygon = useCallback(() => {
+    // We only allow one polygon on the map; if there's an existing one, delete it.
+    const features = draw.getAll().features;
+    if (features.length > 1) {
+      draw.delete(`${features[0].id}`);
+    }
+  }, [draw]);
+
+  const switchToPolygonModeIfFeatureDeleted = useCallback(() => {
+    const features = draw.getAll().features;
+    if (features.length === 0) {
+      draw.changeMode('draw_polygon');
+      setDrawMode('draw_polygon');
+    }
+  }, [draw]);
+
+  const populateGeometry = useCallback(() => {
+    if (!initializedGeometry) {
+      if (boundary) {
+        const featureIds = draw.add(boundary);
+        draw.changeMode('direct_select', { featureId: featureIds[0] });
+        setDrawMode('direct_select');
+      }
+
+      setInitializedGeometry(true);
+    }
+  }, [draw, boundary, initializedGeometry, setDrawMode, setInitializedGeometry]);
+
+  useEffect(() => {
+    if (mapRef?.loaded()) {
+      populateGeometry();
+    } else {
+      mapRef?.on('load', populateGeometry);
+    }
+
+    mapRef?.on('draw.create', removeExistingPolygon);
+    mapRef?.on('draw.delete', switchToPolygonModeIfFeatureDeleted);
+  }, [mapRef, populateGeometry, removeExistingPolygon, switchToPolygonModeIfFeatureDeleted]);
+
+  useControl<GeolocateControl>(
+    () =>
+      new GeolocateControl({
+        fitBoundsOptions: {
+          maxDuration: 1500,
+        },
+        positionOptions: {
+          enableHighAccuracy: true,
+        },
+      })
+  );
+
+  useControl<ScaleControl>(() => new ScaleControl());
+
+  return null;
```

Comment:
It's interesting to define a component and use it for only its hooks, returning `null` at the end. Looks fine to me though—it well encapsulates all the controls logic, separating it from `EditableMap` 👍